pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/validator/DeclarativeValidatorContributor.java

```diff
@@ -74,121 +78,376 @@ public final String validateElement(@Nonnull ModelASTElement element, @CheckForN
         return null;
     }
     
+    @Nonnull
+    public final List<String> validateElementAll(@Nonnull ModelASTElement element, @CheckForNull FlowExecution execution) {
+        String r = validateElement(element, execution);
+        List<String> result = new ArrayList<>();
```

Comment:
No need to allocate a full new `ArrayList`:

`return r != null ? Collections.singeltonList(r) : Collections.emptyList()`
or
`return r != null ? Arrays.asList(r) : Collections.emptyList()`